### ISSUE TITLE
Update node distance map with a volume potential factor

### ISSUE THREAD

**Issue by JeffreyDevloo** [openvstorage/framework#2094]
### Enhancement description
HA is currently done by the edge client which checks the node distance map.
When volumedriver instances can no longer hold any more volumes: it should not be eligible HA destination

**Comment by wimpers** [comment]
@redlicha does the voldrv still accept the new volume or does it fail straight away?

**Comment by redlicha** [comment]
A "filled up" voldrv will become the new owner (if instructed by HA / the API), but actually starting the volume will fail due to resource shortage (most likely candidate: SCO cache cannot guarantee progress b/c there's not enough room to cover a TLog worth of SCOs).

**Comment by wimpers** [comment]
@redlicha any reason why we not block the transfer of the ownership so the edge knows it selected the wrong one? I assume this has to do with the fact that you can move a volume but not start it and only started volumes are taking into account for volume potential? Maybe we can block the ownership move (leaving a force method)?

**Comment by redlicha** [comment]
No, it's not about non-running volumes but the underlying assumption so far that HA will not be attempted to nodes that can't host more volumes based on the node distance map.

Adhoc I think it's possible to add an extra check of the local volume potential to voldrv before transferring (grabbing) the ownership (NB: the check is expensive in that it has to fetch the volume config from the backend. This should not impact the "happy" failover path where there are sufficient resources as that can be reworked accordingly, but the case where the voldrv cannot host more volumes).

**Comment by wimpers** [comment]
> ... underlying assumption so far that HA will not be attempted to nodes that can't host more volumes based on the node distance map...

In my opinion this is abusing the node distance map.
